RepoDb.Core/RepoDb/ClassProperty.cs

```diff
@@ -90,7 +91,10 @@ public PrimaryAttribute GetPrimaryAttribute()
                 return m_primaryAttribute;
             }
             m_isPrimaryAttributeWasSet = true;
-            return m_primaryAttribute = PropertyInfo.GetCustomAttribute(typeof(PrimaryAttribute)) as PrimaryAttribute;
+
+            return m_primaryAttribute = PropertyInfo.GetCustomAttribute(typeof(KeyAttribute)) != null
```

Comment:
I am thinking that when it comes to priority/precedence, the `PrimaryAttribute` is must more higher over `KeyAttribute`.

**Why?** Most likely, a lot of users already using the `[Map]` and in order for us not to affect the efficiency and performance, we should make it higher instead.

But TBH, I do not know for the new users. Will they use the `KeyAttribute` over the `PrimaryAttribute`? I think you can answer this :).

Reply:
This concern is one of those coin toss type issues until you have actual measurements.  For my needs, the precedence doesn't really matter to me.  

I discovered RepoDB from the Reddit C# forum and was intrigued by performance measures compared to Dapper.  So performance is important.  I have a set of libraries that I use in my application development and my data layer library wraps Dapper to fill in the missing features such as SQL bulk operations and interpreting Linq predicates to SQL.

What impressed me with RepoDB is that I could throw away all the support code I needed for Dapper.  Thus, I think RepoDB will attract a large audience of developers from both Dapper and EF.  (I never really liked EF because it's too heavy and too complex trying to do everything from the client).

Dapper/EF developers moving to RepoDB, IMO, will be more concern about migrating their Dapper/EF POCOs.  So having the attribute support is a convenience for them and will aid RepoDB's adoption.   This is why I couldn't wait any longer for @lauw to submit his PR because that was holding me up.

It is hard to say how RepoDB is going to be used and it may be one of those issues you'll have to monitor to gauge actual usage and performance impact.  For now, what should take higher priority is that this change does minimal to no impact your current performance measures.  So from that perspective, the precedence should favor the current RepoDB model where Primary & Map are higher then Key & Table, Column attributes.

Cheers.

Reply:
P.S. Let me know if you agree.

Reply:
> For now, what should take higher priority is that this change does minimal to no impact your current performance measures.

Valid reason, and besides it is on compiler level of RepoDb which will only be executed once. Then I have to update the docs and some tests when it comes to this priority/precedence.

Thanks for this very great PR with test suites :)

Reply:
I'll resubmit the PR with the fixes